File: sach/de3386.py
//DE3386.PY - Tran Huu Nam - hsgtin.vn - 2024
import sys
f = [0] * 81
s = [0] * 81
def main():
    input = sys.stdin.read
    data = input().split()
    m = int(data[0])
    f[0] = 0
    f[1] = 1
    s[0] = 0
    s[1] = 1
    for i in range(2, 81):
        f[i] = f[i - 1] + f[i - 2]
        s[i] = s[i - 1] + f[i]
    for i in range(1, m + 1):
        n = int(data[i])
        print(s[n], end=" ")
if __name__ == "__main__":
    main()

File: sach/de3386.cpp
//DE3386.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <bits/stdc++.h>
using namespace std;
long long f[81], s[81];
int main() {
    ios_base::sync_with_stdio(false); cin.tie(NULL);
    int m,n;
    cin>>m;
    f[0]=0; f[1] = 1;
    s[0]=0;  s[1]=1;
    for(int i=2;i<=80;i++) {
        f[i] = f[i-1] + f[i-2];
        s[i] = s[i-1] + f[i];
    }
    for(int i=1;i<=m;i++){
      cin>>n;
      cout<<s[n]<<" ";
    }
}

